README.md

```diff
@@ -141,78 +146,93 @@ Generated spec:
-          "type": [
-            "null",
-            "string"
-          ]
+          "type": ["null", "string"]
         },
         "lastName": {
-          "type": [
-            "null",
-            "string"
-          ]
+          "type": ["null", "string"]
         }
       }
     },
     "Event": {
       "type": "object",
       "additionalProperties": false,
       "properties": {
-        "Type": {
-          "type": [
-            "null",
-            "string"
-          ]
+        "type": {
+          "type": ["null", "string"]
         }
       }
     }
   }
 }
 ```
 
-Generated TypeScript code: 
+Generated TypeScript code:
 
 ```typescript
-import { HubConnection, IStreamResult } from "@aspnet/signalr"
+import { HubConnection, IStreamResult } from '@microsoft/signalr';
 
 export class ChatHub {
-    constructor(private connection: HubConnection) {
-    }
+  constructor(private connection: HubConnection) {}
 
-    send(message: string): Promise<void> {
-        return this.connection.invoke('Send', message);
-    }
+  send(message: string): Promise<void> {
+    return this.connection.invoke('Send', message);
+  }
 
-    addPerson(person: Person): Promise<void> {
-        return this.connection.invoke('AddPerson', person);
-    }
+  addPerson(person: Person): Promise<void> {
+    return this.connection.invoke('AddPerson', person);
+  }
 
-    getEvents(): IStreamResult<Event> {
-        return this.connection.stream('GetEvents');
-    }
+  getEvents(): IStreamResult<Event> {
+    return this.connection.stream('GetEvents');
+  }
 
-    registerCallbacks(implementation: IChatHubCallbacks) {
-        this.connection.on('Welcome', () => implementation.welcome());
-        this.connection.on('Send', (message) => implementation.send(message));
-    }
+  onWelcome(func: () => void): void {
+    this.connection.on('Welcome', func);
+  }
 
-    unregisterCallbacks(implementation: IChatHubCallbacks) {
-        this.connection.off('Welcome', () => implementation.welcome());
-        this.connection.off('Send', (message) => implementation.send(message));
-    }
+  unregisterWelcome(func: () => void): void {
+    this.connection.off('Welcome', func);
+  }
+  onSend(func: (message: string) => void): void {
+    this.connection.on('Send', func);
+  }
+
+  unregisterSend(func: (message: string) => void): void {
+    this.connection.off('Send', func);
+  }
+  onPersonAdded(func: (person: Person) => void): void {
+    this.connection.on('PersonAdded', func);
+  }
+
+  unregisterPersonAdded(func: (person: Person) => void): void {
+    this.connection.off('PersonAdded', func);
+  }
+
+  registerCallbacks(implementation: IChatHubCallbacks) {
+    this.connection.on('Welcome', implementation.welcome);
+    this.connection.on('Send', implementation.send);
+    this.connection.on('PersonAdded', implementation.personAdded);
+  }
+
+  unregisterCallbacks(implementation: IChatHubCallbacks) {
+    this.connection.off('Welcome', implementation.welcome);
```

Comment:
As i mentioned in other PRs this will not work because "this" in the welcome call will not be "implementation" (I think)

Reply:
Ok good point i have reverted my changes for the register and unregister callback methods

Reply:
I think we need to internally store the registered lambdas because the off-methods will "dynamically" new lambda which does not have the same identity (ie not ===) as the registered one - or am I wrong?

Reply:
I think that the best approach might be to write some unit tests around this topic so we can make sure everything works as intended.